Explorer/Assets/DCL/ExplorePanel/ExplorePanelInputHandler.cs

```diff
@@ -1,35 +1,61 @@
+using DCL.ExplorePanel.Components;
 using DCL.UI;
 using MVC;
 using System;
+using System.Collections.Generic;
 using UnityEngine.InputSystem;
 
-namespace DCL.ExplorePanel
+namespace DCL.ExplorePanel.Component
 {
-    public class ExplorePanelInputHandler : IDisposable
+    public class ExplorePanelInputHandler : IDisposable, IExplorePanelEscapeAction
     {
         private readonly DCLInput dclInput;
         private readonly IMVCManager mvcManager;
+        private readonly LinkedList<Action<InputAction.CallbackContext>> escapeActions = new ();
+        private readonly bool includeCameraReel;
         private static ExploreSections lastShownSection;
 
-        public ExplorePanelInputHandler(DCLInput dclInput, IMVCManager mvcManager)
+        public ExplorePanelInputHandler(DCLInput dclInput, IMVCManager mvcManager, bool includeCameraReel)
         {
             this.dclInput = dclInput;
             this.mvcManager = mvcManager;
+            this.includeCameraReel = includeCameraReel;
             lastShownSection = ExploreSections.Navmap;
             RegisterHotkeys();
         }
 
-        public void Dispose()
+        public void RegisterEscapeAction(Action<InputAction.CallbackContext> action)
```

Comment:
Although is not necessary, but this design could be improved by a command pattern, something we did with the navmap. Basically make a command class for each type of navigation and register it in a stack. Then you execute it backwards whenever you need to go back.
- Command interface: https://github.com/decentraland/unity-explorer/blob/bf927d678e249cb3e4efcc3e102b83dae3085c9e/Explorer/Assets/DCL/Navmap/INavmapCommand.cs#L7
- Implementation of the back: https://github.com/decentraland/unity-explorer/blob/bf927d678e249cb3e4efcc3e102b83dae3085c9e/Explorer/Assets/DCL/Navmap/NavmapCommandBus.cs#L61
- Example of a command: https://github.com/decentraland/unity-explorer/blob/bf927d678e249cb3e4efcc3e102b83dae3085c9e/Explorer/Assets/DCL/Navmap/ShowPlaceInfoCommand.cs#L41

In you case the commands could be simplified.

Reply:
Although I really like the approach, I don't think here is necessary as I just need to handle the effect that the escape button has while being in the Explore Panel. There is no need for custom actions/behaviors, therefore it would be a huge structure with no visible gain, in my opinion.